pkgs/applications/misc/st/xst.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchurl, pkgconfig, libX11, ncurses, libXext, libXft , fontconfig
+, patches ? [], extraLibs ? []}:
+
+with stdenv.lib;
+
+let
+  version = "0.7.1";
+  name = "xst-${version}";
+  patches' = if patches == null then [] else patches;
```

Comment:
Actually, I'm not really sure what `?` operator does, even after reading the nix manual. Are you saying that `patches ? []` is effectively the same as this `if` statement? Because I was just copying what they did in the `default.nix` file in this directory.

(Though I've been using nixos as a user for a few years, I've never learned nix lang that much and still very new to it. 😕)